apps/frontend/core-dapp/e2e/withdraw.spec.ts

```diff
@@ -0,0 +1,84 @@
+/* eslint-disable testing-library/prefer-screen-queries,jest/no-done-callback */
+import { expect, test } from '@playwright/test'
+import { NETWORKS } from 'prepo-constants/lib/src'
+import { Wallet } from 'ethers'
+import { formatEther } from 'ethers/lib/utils'
+import { addBalanceToWallet, spawnTenderlyRpc } from './utils/tenderly'
+import { increaseDepositLimits, increaseWithdrawLimits } from './utils/limits'
+import { acceptTerms, connectWalletToPlaywright } from './utils/playwright'
+import { TestIds } from '../src/components/TestId'
+
+test('The user receives at least the minimum amount advertised in the UI, after slippage', async ({
+  page,
+}) => {
+  const wallet = Wallet.createRandom()
+
+  const rpcUrl = await spawnTenderlyRpc()
+
+  // The front-end doesn't subscribe to changes in limits because they are not
+  // expected to change frequently.
+  // Increase the limits before loading the front-end.
+  await increaseDepositLimits({ rpcUrl })
+  await increaseWithdrawLimits({ rpcUrl })
+
+  const walletConnect = await connectWalletToPlaywright({
+    page,
+    rpc: {
+      chainId: NETWORKS.arbitrumOne.chainId,
+      url: rpcUrl,
+    },
+    wallet,
+  })
+
+  await addBalanceToWallet({
+    address: wallet.address,
+    provider: await walletConnect.provider,
+  })
+
+  await acceptTerms({ page })
+
+  await page.getByRole('link', { name: 'Portfolio' }).click()
+  await page.getByRole('link', { name: 'Deposit →' }).click()
+  await page.getByPlaceholder('0').fill('100')
+  await page.getByRole('button', { name: 'Deposit', exact: true }).click()
+  await walletConnect.waitForTxHash()
+
+  const balanceBefore = await (await walletConnect.provider).getBalance(wallet.address)
```

Comment:
ditto, code would be more readable if we have provider on top?

```suggestion
  const balanceBefore = await provider.getBalance(wallet.address)
```